OpenRA.Game/Support/PerfTickLogger.cs

```diff
@@ -13,36 +13,41 @@
 
 namespace OpenRA.Support
 {
-	public sealed class PerfTickLogger
+	public static class PerfTickLogger
 	{
-		readonly DebugSettings settings = Game.Settings.Debug;
-		readonly long threshold = PerfTimer.LongTickThresholdInStopwatchTicks;
-		long start;
-		long current;
-		bool enabled;
+		public const long TickDisabled = 0L;
 
-		long CurrentTimestamp => enabled ? Stopwatch.GetTimestamp() : 0L;
+		static float longTickThresholdMs = Game.Settings.Debug.LongTickThresholdMs;
+		static long longTickThresholdTicks = PerfTimer.MillisToTicks(Game.Settings.Debug.LongTickThresholdMs);
 
-		public void Start()
+		/// <returns>TickDisabled if performance logging is disabled.</returns>
+		public static long Start()
 		{
-			enabled = settings.EnableSimulationPerfLogging;
-			start = CurrentTimestamp;
+			var settings = Game.Settings.Debug;
+			if (!settings.EnableSimulationPerfLogging)
+				return TickDisabled;
+
+			// TODO: Let settings notify listeners on changes
+			if (longTickThresholdMs != settings.LongTickThresholdMs)
+			{
+				longTickThresholdMs = Game.Settings.Debug.LongTickThresholdMs;
+				longTickThresholdTicks = PerfTimer.MillisToTicks(longTickThresholdMs);
+			}
+
+			return Stopwatch.GetTimestamp();
 		}
 
-		public void LogTickAndRestartTimer(string name, object item)
+		/// <returns>TickDisabled if performance logging is disabled.</returns>
+		public static long LogTickAndRestartTimer(long startTick, string name, object item)
```

Comment:
This nolonger resets the timer so it should be renamed

Reply:
Well, it never did. Functionally it did not change. Also in the past it would only update current  - not reset any stopwatches or times - which now only is returned.

Do you have a proposal for a better name? `LogTickAndReturnCurrentTime`?

Reply:
It did reset in the sense that it the start time was set.

I'd say both the `Start` and this function need name changes. They could be changed to something like `GetCurrentTime` and `LogPerf`